src/VisualStudio/Core/Def/Snippets/AbstractSnippetExpansionClient.cs

```diff
@@ -221,6 +282,45 @@ public int FormatSpan(IVsTextLines pBuffer, VsTextSpan[] tsInSurfaceBuffer)
 
             var formattingSpan = CommonFormattingHelpers.GetFormattingSpan(SubjectBuffer.CurrentSnapshot, snippetTrackingSpan.GetSpan(SubjectBuffer.CurrentSnapshot));
 
+            // If the snippet spans more than one line and a specific line ending is expected, normalize the line
+            // endings inside the snippet span before continuing.
+            if (newLineEnding is not null && tsInSurfaceBuffer[0].iEndLine > tsInSurfaceBuffer[0].iStartLine)
```

Comment:
Should we be wrapping this additional edit with an undo transaction since you're now doing multiple edits at once?

Reply:
I believe no. This is working around a bug in the snippet engine where it fails to normalize line endings from some (but not all) snippets. This only makes a change in cases where a snippet was incorrectly processed, and the result is considered part of the original snippet insertion.